### PR TITLE
Testing

### LINKED ISSUE #15: Setup Testing
Setup a basic testing structure for the various routes available.

### PR DESCRIPTION
**Pull request by yesoer**
# ATTENTION :
handlers package and router.go (from main package) have been combined as api package in [da28c8b](https://github.com/PPSKSY-Cluster/backend/pull/17/commits/da28c8bf5fb4dbc8a86b45528296b66f97f867f2). 
Also  in [c4df05a](https://github.com/PPSKSY-Cluster/backend/pull/17/commits/c4df05a7127afc244dab0bf8ff138881466be1a8) a conditionally built part of the db package was introduced to allow for database dropping if the tag "test" was supplied. 
See Commit messages for reasoning.

------------------------------------------------------------------------------------------------------------------------

resolves #15 
- using testing and stretchr/testify
- providing a more generalized testing setup (TestReq struct and a function that can execute them) see [6718353](https://github.com/PPSKSY-Cluster/backend/pull/17/commits/6718353c45d9a7c83c9c8019b7c31b014b6097d8)
- tests 
  - api.userListHandler or GET /api/users
  - api.userCreateHandler or POST /api/users
  - api.userEditHandler or PUT /api/users/:id
  - api.userDetailHandler or GET /api/users/:id
  - api.userDeleteHandler or DELETE /api/users/:id 
  - api.loginHandler or POST /api/login
  - api.cResourceListHandler or GET /api/cresources
  - api.cResourceCreateHandler or POST /api/cresources
  - api.cResourceEditHandler or PUT /api/cresources/:id
  - api.cResourceDetailHandler or GET /api/cresources/:id
  - api.cResourceDeleteHandler or DELETE /api/cresources/:id 
  
Should be merged after PR #13 (Done, all conflicts resolved).

### CODE REVIEW

File: tests/test.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+	"github.com/gofiber/fiber/v2"
+	"github.com/joho/godotenv"
+	"github.com/stretchr/testify/assert"
+)
+
+const TIMEOUT int = -1
+
+func setupTestApplication() (*fiber.App, error) {
+	// Server setup : load env file, setup db, setup router
+	if err := godotenv.Load("./.env"); err != nil {
```
Comment (vDawgg) [inline tests/test.go:23]: Do we plan to use different env files for testing and running? If so this should be stated in the README

File: README.md [hunk 2]
```diff
@@ -20,6 +20,15 @@ To compile it  run
 ```
 go build -o ./build/server main.go router.go
 ```
```
Comment (vDawgg) [inline README.md:22]: As main and router live in different directories now the run and build instructions should be adapted accordingly (Why is router run anyways when main alone seems sufficient?)

**Review by vDawgg — COMMENTED** [review]
Are the tests actually completing succesfully on your side? Dont know if something is wrong with my db setup but all the tests are failing for me...
Otherwise I like the structure of the testing suite, pretty clean!

File: tests/test.go (hunk 1 above)
Comment (yesoer) [reply tests/test.go:23]: Ah yes it seems like I accidentaly commited that [here](https://github.com/PPSKSY-Cluster/backend/pull/28/commits/16a42874968227ed1a693c37401fdcfc7c80653f). I will try to clean that up. The main reason is that tests tend to clutter databases, so one would probably want to use a different one for testing, that can be dropped after each test (what testInterface.go is for).

File: README.md (hunk 2 above)
Comment (yesoer) [reply README.md:22]: Yes you're right we only need go run main.go etc.

**Comment by yesoer** [comment]
> Are the tests actually completing succesfully on your side? Dont know if something is wrong with my db setup but all the tests are failing for me... Otherwise I like the structure of the testing suite, pretty clean!

Yes everything works fine for me. Could you provide the error message(s) ?

**Comment by vDawgg** [comment]
> 

Nvm, seems to be a setup problem on my side. Just tested on another machine and works perfectly now.

**Pull request merged**